sborenst/ansible_agnostic_deployer
# fix zabbix-client role and create new zabbix roles

The motivation for this commit is to have zabbix-client as a generic role and
additional roles for each type of host (OCP Node, master, provisionner, etc) .

This has been tested against configs/ocp-workshop.

zabbix-client:
- add README.adoc
- fix:
  - add zabbix repository to be able to install zabbix-agent
  - do not restart iptables-save (see comment inline in code)
- improvements:
  - split main.yml into several yaml files with includes
  - add _optional_ auto_registration. 3 input params:
    - zabbix_hostname
    - zabbix_auto_registration_keyword
    - zabbix_auto_registration_pass
  - by default auto-registration is disable, see zabbix-client/README.adoc

new roles:
- add zabbix-client-openshift-node:
  - create README.adoc
  - install necessary stuff for 'template OCP Node' to work
  - Add sudoers, Zabbix UserParameter, ...
- add zabbix-client-openshift-master:
  - create README.adoc
  - create zabbix Service Account
  - fetch and set zabbix_token
  - try to fetch and set hawkular_route fact
- enable these new role in ocp-workshop/post-software.yml.
  - can still be skipped with --skip-tags=install_zabbix

How To test with ocp-workshop:
- without auto-registration: tested, should work with no change
- with auto-registration:
  - just set zabbix_auto_registration_pass in env_secret_vars.yml
    (see value in mon.opentlc.com, Configuration/Actions)
